doors_dashboards/components/meteogram.py

```diff
@@ -208,18 +229,94 @@ def register_callbacks(self, app: dash.Dash, component_ids: List[str]):
+                    component_data.update(component_state_data)
+                else:
+                    component_data = component_state_data
+                date_string = component_data.get("date", "") if component_data.get(
+                    "date") else None
+                forecast_value = component_data.get("meteogram_type",
+                                                    OPTIONS[0]["value"])
+            else:
+                date_string = None
+                forecast_value = OPTIONS[0]["value"]
 
             return meteo_label, self._get_meteogram_image(
                 meteo_lon, meteo_lat, date_string, forecast_value
             )
+
+        @app.callback(
+            Output(TEMP_STORE_ID, "data", allow_duplicate=True),
+            [Input(dropdown_id, 'n_clicks_timestamp')
+             for dropdown_id in list(OPTION_TO_ID.values())],
+            prevent_initial_call=True
+        )
+        def selector_to_component_store(*timestamps):
+            clicked_index = timestamps.index(
+                    max(t for t in timestamps if t is not None))
+            meteogram_drp_option_label = OPTIONS[clicked_index]['value']
+
+            temp_data = {
+                'meteogram_type': meteogram_drp_option_label,
+            }
+
+            return temp_data
+
+        @app.callback(
+            Output(METEOGRAM_CHOOSER_ID, "label"),
+            Input(COMPONENT_STORE_ID, "data"),
+            prevent_initial_call=True
+        )
+        def component_store_to_drp_label(component_data):
+            if component_data is None:
+                return no_update
+            selected_label = component_data.get("meteogram_type", OPTIONS[0]["value"])
+            return selected_label
+
+        @app.callback(
+            Output(COMPONENT_STORE_ID, "data"),
+            Input(METEOGRAM_DATE_PICKER_ID, 'date'),
+            State(COMPONENT_STORE_ID, 'data'),
+            prevent_initial_call=True
+        )
+        def datepicker_to_component_store(date_value, component_data):
+            if component_data is not None:
+                component_data = component_data or {}
+            date_string = date.fromisoformat(date_value). \
+                strftime('%Y-%m-%dT00:00:00Z') if date_value else None
+            selected_date = {
+                'date': date_string
+            }
+            component_data.update(selected_date)
+            return component_data
+
+        @app.callback(
+            Output(COMPONENT_STORE_ID, "data", allow_duplicate=True),
+            Input(TEMP_STORE_ID, 'data'),
+            State(COMPONENT_STORE_ID, 'data'),
+            prevent_initial_call=True
+        )
+        def datepicker_to_component_store(temp_data, component_data):
+            if temp_data is None:
+                return  no_update
+            if component_data is not None:
+                component_data = component_data or {}
+            else:
+                component_data = {}
```

Comment:
```suggestion
            component_data = component_data or {}
```

This is doing the same thing in just one line.